packages/dev/core/src/DeviceInput/InputDevices/deviceSourceManager.ts

```diff
@@ -91,12 +78,18 @@ export class DeviceSourceManager implements IDisposable, IObservableManager {
         this._engine._deviceSourceManager._refCount++;
 
         // Observables
-        this.onDeviceConnectedObservable = new Observable<DeviceSource<DeviceType>>((observer) => {
-            this.getDevices().forEach((device) => {
-                this.onDeviceConnectedObservable.notifyObserver(observer, device);
-            });
+        this.onDeviceConnectedObservable = new Observable((observer) => {
+            for (const devices of this._devices) {
+                if (devices) {
+                    for (const device of devices) {
+                        if (device) {
+                            this.onDeviceConnectedObservable.notifyObserver(observer, device as DeviceSourceType);

```

Comment:
I guess this cast is fall out from removing the type info for `_devices`?

Reply:
I believe that it's forcing us to cast because DeviceSource<DeviceType> isn't being seen as equal to DeviceSourceType